### PR TITLE
Improve behaviour for invalid CLI options

### LINKED ISSUE #3602: Improve behaviour for invalid CLI options
> Clearly describe the bug

1. Run `stylelint --unknown` command in terminal.
2. Print help, but stylelint doesn't tell me that I gave an invalid option.
3. Return exit code `2`.

> Which rule, if any, is the bug related to?

n/a

> What CSS is needed to reproduce the bug?

n/a

> What stylelint configuration is needed to reproduce the bug?

n/a

> Which version of stylelint are you using?

`9.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --unknown`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

n/a

> What did you expect to happen?

I think that stylelint should tell users kindly that they gave invalid options, instead of printing help.

For example, in case of ESLint:

```
$ eslint --unknown
Invalid option '--unknown' - perhaps you meant '--ignore'?
```

> What actually happened (e.g. what warnings or errors did you get)?

stylelint doesn't tell me that I gave an invalid option.

```
$ stlyelint --unknown
...
      Useful when setting "defaultSeverity" to "warning" and expecting the
      process to fail on warnings (e.g. CI build).

    --version, -v

      Show the currently installed version of stylelint.
```

### PR DESCRIPTION
**Pull request by ybiquitous**
This change displays the user-friendly message(s) instead of help when invalid options are given.

For example:

```
$ stylelint --foo
Invalid option "--foo". Did you mean "--fix"?
```

image

Change summary:

- Add package: [`leven`](https://github.com/sindresorhus/leven)
- Add utility method: `checkInvalidCLIOptions()`

> Which issue, if any, is this issue related to?

Closes #3602 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

**Comment by jeddy3** [comment]
> (Perhaps it is better to stop suggestion for shortcut options?)

SGTM. Even with just the non-shortcut suggestions, this is a great improvement. We can revisit an approach to shortcuts later down the line.

Do you want to update this PR?

**Comment by ybiquitous** [comment]
> Do you want to update this PR?

Yes. I want to stop suggestion for shortcuts.

File: lib/utils/checkInvalidCLIOptions.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    suggestion = sameLength.find(option => leven(option, invalid) <= 2);
+  }
+  if (!suggestion) {
+    suggestion = all.find(option => leven(option, invalid) <= 2);
+  }
+  if (!suggestion) {
+    suggestion = all.find(option => leven(option, invalid) <= 3);
+  }
+  return suggestion;
+};
```
Comment (ybiquitous) [inline lib/utils/checkInvalidCLIOptions.js]: This `suggest` algorithm seems not good to me... 😓 

```
$ ./bin/stylelint.js --hel
Invalid option "--hel". Did you mean "-h"?
```

### Ideal

```
$ ./bin/stylelint.js --hel
Invalid option "--hel". Did you mean "--help"?
```

Comment (jeddy3) [reply lib/utils/checkInvalidCLIOptions.js]: Yep, the latter is better.

**Comment by ybiquitous** [comment]
I made some improvements via commit a3e96e7 😃 

- No suggestion for shortcut options
- Better suggestion algorithm

**Comment by ybiquitous** [comment]
Why did appveyor build fail? 

https://ci.appveyor.com/project/stylelint/stylelint/build/6193/job/2rtxpmv9qv0wa964#L344

image

**Comment by hudochenkov** [comment]
@ybiquitous no one knows we it's happening from time to time. We have some problems with ignoring parts of stylelint.

I restarted the build. It usually helps. I believe you can also restart it yourself next time if you sign in to AppVeyor with your Github account.

**Comment by ybiquitous** [comment]
Thanks a lot!

**Comment by jeddy3** [comment]
> -   Added: suggestions for invalid CLI options ([#3622](https://github.com/stylelint/stylelint/pull/3622)).

**Comment by ybiquitous** [comment]
@jeddy3 Thanks a lot! I am looking forward to the next release! 😆

**Pull request merged**